Assets/Scripts/Modules/Combat/ICombatModule.cs

```diff
@@ -1,8 +1,19 @@
 ﻿/*
  * Authors: Ian Macpherson, Isaiah Mann
- * Description: An interface to handle the turn logic
+ * Description: An interface to handle the combat logic
  */
 
 public interface ICombatModule {
 
+	bool IsTargetInRange (IUnit unit);
+
+	bool IsTargetAdjacent (IUnit unit);
+
+	void MeleeAttack (IUnit unit);
+
+	void RangedAttack (IUnit unit);
+
+	void FleeAttempt (IStatModule playerstats, IUnit unit);
+
+	void KillMonster (IUnit unit);
```

Comment:
That makes since, the flee mechanic corresponds to players death but that's not clear without understanding the full context of the game so will do.